core/src/main/java/org/opensearch/sql/expression/aggregation/NamedAggregator.java

```diff
@@ -64,6 +66,7 @@ public NamedAggregator(
     super(delegated.getFunctionName(), delegated.getArguments(), delegated.returnType);
     this.name = name;
     this.delegated = delegated;
+    this.condition = delegated.condition != null ? delegated.condition : null;
```

Comment:
Is this same as `this.condition = delegate.condition`? Or you actually wanted to check if `delegated` is null?

Reply:
It should be the same with `this.condition = delegate.condition`. Thanks!